fem/fespace.hpp

```diff
@@ -1012,6 +1030,113 @@ class QuadratureInterpolator
                       const int eval_flags);
 };
 
+/** @brief A class that performs interpolation from an E-vector to quadrature
+    point values and/or derivatives (Q-vectors) on the faces. */
+/** An E-vector represents the element-wise discontinuous version of the FE
+    space and can be obtained, for example, from a GridFunction using the
+    Operator returned by FiniteElementSpace::GetElementRestriction().
+
+    The target quadrature points in the elements can be described either by an
+    IntegrationRule (all mesh elements must be of the same type in this case) or
+    by a QuadratureSpace. */
+class FaceQuadratureInterpolator
```

Comment:
It would be also good to start to avoid putting things more in `fespace.*`, shouldn't we open a new file for the `QuadratureInterpolator`?

Reply:
I'm okay with it, @v-dobrev @tzanio ?

Reply:
I'm fine with it, as long as it is an independent class that makes sense on its own.

Reply:
Moving the `*Interpolator` classes to a new file sounds good to me -- it will probably speed up the build a little. :smile:

Reply:
done.